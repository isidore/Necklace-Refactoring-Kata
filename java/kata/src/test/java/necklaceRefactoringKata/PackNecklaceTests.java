package necklaceRefactoringKata;


import necklaceRefactoringKata.jewellery.Jewel;
import necklaceRefactoringKata.jewellery.Necklace;
import necklaceRefactoringKata.jewellery.NecklaceType;
import org.approvaltests.Approvals;
import org.approvaltests.JsonApprovals;
import org.junit.jupiter.api.Test;

class PackNecklaceTests {
    @Test
    void test_AmberBeadsNecklace_IsPackedInTree() {
        var item = new Necklace(Jewel.Pearl, NecklaceType.Beads);
        var storage = new JewelleryStorage();

        Packer.packNecklace(item, storage);
        Packer.packNecklace(new Necklace(Jewel.Diamond, NecklaceType.Beads), storage);
        Packer.packNecklace(new Necklace(Jewel.Amber, NecklaceType.Chain), storage);

        JsonApprovals.verifyAsJson(storage);
        // TODO: check if packed in right container
    }
}
