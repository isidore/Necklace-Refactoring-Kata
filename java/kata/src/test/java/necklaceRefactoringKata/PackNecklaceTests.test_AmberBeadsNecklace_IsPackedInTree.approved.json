{
  "box": {
    "ringCompartment": [],
    "topShelf": [
      {
        "type": "Chain",
        "stone": "Amber"
      }
    ],
    "mainSection": []
  },
  "tree": [
    {
      "type": "Beads",
      "stone": "Pearl"
    }
  ],
  "travelRoll": [],
  "safe": [
    {
      "type": "Beads",
      "stone": "Diamond"
    }
  ],
  "dresserTop": []
}